src/graphapi/matmul.cpp

```diff
@@ -104,5 +105,310 @@ void BackendMatmulDescriptor::getAttribute(miopenBackendAttributeName_t attribut
     }
 }
 
+OperationMatmulBuilder& OperationMatmulBuilder::setA(Tensor* A)
+{
+    mOperationMatmul.mA = checkPtr(A);
+    if(mOperationMatmul.mA->getDimensions().size() < 2)
+        MIOPEN_THROW(miopenStatusBadParm);
+    aSet = true;
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setB(Tensor* B)
+{
+    mOperationMatmul.mB = checkPtr(B);
+    if(mOperationMatmul.mB->getDimensions().size() < 2)
+        MIOPEN_THROW(miopenStatusBadParm);
+    bSet = true;
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setC(Tensor* C)
+{
+    mOperationMatmul.mC = checkPtr(C);
+    if(mOperationMatmul.mC->getDimensions().size() < 2)
+        MIOPEN_THROW(miopenStatusBadParm);
+    cSet = true;
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setBatchCount(int64_t count)
+{
+    mOperationMatmul.mBatchCount = count;
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setGemmMOverride(Tensor* overrideTensor)
+{
+    mOperationMatmul.mGemmMOverride = checkPtr(overrideTensor);
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setGemmNOverride(Tensor* overrideTensor)
+{
+    mOperationMatmul.mGemmNOverride = checkPtr(overrideTensor);
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setGemmKOverride(Tensor* overrideTensor)
+{
+    mOperationMatmul.mGemmKOverride = checkPtr(overrideTensor);
+    return *this;
+};
+OperationMatmulBuilder& OperationMatmulBuilder::setMatmulDescriptor(Matmul* mMatmul)
+{
+    mOperationMatmul.mMatmul = checkPtr(mMatmul);
+    matmulSet                = true;
+    return *this;
+}
+
+OperationMatmul OperationMatmulBuilder::build()
+{
+    if(!aSet || !bSet || !cSet || !matmulSet)
+        MIOPEN_THROW(miopenStatusBadParm);
+
+    int aDimensionsCount = mOperationMatmul.mA->getDimensions().size();
+    int bDimensionsCount = mOperationMatmul.mB->getDimensions().size();
+    int cDimensionsCount = mOperationMatmul.mC->getDimensions().size();
+
+    if(aDimensionsCount != bDimensionsCount || bDimensionsCount != cDimensionsCount)
+        MIOPEN_THROW(miopenStatusBadParm);
+
+    int Am = mOperationMatmul.mA->getDimensions().end()[-2];
+    int An = mOperationMatmul.mA->getDimensions().end()[-1];
+
+    int Bn = mOperationMatmul.mB->getDimensions().end()[-2];
+    int Bk = mOperationMatmul.mB->getDimensions().end()[-1];
+
+    int Cm = mOperationMatmul.mB->getDimensions().end()[-2];
+    int Ck = mOperationMatmul.mB->getDimensions().end()[-1];
```

Comment:
```suggestion
    int An = mOperationMatmul.mA->getDimensions().begin()[aDimensionsCount-1];

    int Bn = mOperationMatmul.mB->getDimensions().begin()[bDimensionsCount-2];
    int Bk = mOperationMatmul.mB->getDimensions().begin()[bDimensionsCount-1];

    int Cm = mOperationMatmul.mC->getDimensions().begin()[cDimensionsCount-2];
    int Ck = mOperationMatmul.mC->getDimensions().begin()[cDimensionsCount-1];
```
Same as above: `cppcheck` doesn't allow to dereference `end()` iterators.
And more serious, could you check the last 2 lines: should there be `mB` or `mC`?

Reply:
mC, thanks

Reply:
fixed